cosipy/image_deconvolution/allskyimage.py

```diff
@@ -151,9 +151,11 @@ def set_values_from_parameters(self, parameter):
         if algorithm_name == "flat":
             unit = u.Unit(algorithm_parameter['unit'])
             for idx, value in enumerate(algorithm_parameter['value']):
-                self[:,idx] = value * unit
+                self[:,idx] = value * unit      # TODO: Why is this indexed as [:, idx] instead of [idx, :]?
```

Comment:
> Why is this indexed as [:, idx] instead of [idx, :]?

It is because the order of axes is defined as (lb, Ei). It follows the axes order of the full detector response.